pkg/reaper/nodereaper/helpers.go

```diff
@@ -209,6 +212,46 @@ func (ctx *ReaperContext) publishEvent(namespace string, event *v1.Event) error
 	return nil
 }
 
+func obtainReapLock(ddbAPI dynamodbiface.DynamoDBAPI, nodeName, instanceID, nodeType string) error {
+	log.Infof("obtaining lock for a %s node %s (%s)", nodeType, nodeName, instanceID)
+
+	timestamp := time.Now().Format(time.RFC3339)
+
+	lock := LockRecord{
+		LockType:   nodeType,
+		NodeName:   nodeName,
+		InstanceID: instanceID,
+		NodeType:   nodeType,
+		CreatedAt:  timestamp,
+		// TODO: make configurable
+		ExpiresAt: time.Now().Unix() + int64(30), // expire locks automatically
+	}
+
+	return lock.obtainLock(ddbAPI)
+}
+
+func (l LockRecord) obtainLock(ddbAPI dynamodbiface.DynamoDBAPI) error {
```

Comment:
Could we implement with the same interface as `sync.Mutex`? Just to give a more friendly API: https://pkg.go.dev/sync#Mutex

Reply:
Hmm, not sure how useful that would be, because that interface doesn't allow errors :/ And `Lock`, I guess, is supposed to be synchronous per
> If the lock is already in use, the calling goroutine blocks until the mutex is available.